Level up rewrite

Vesly01 opened the pull request:
- Save level that you promote at in BWL->squares moved 
- Do not save squares moved (I don't think this is used anywhere) 
- Improve accuracy of fixed growths by accounting for the level you promote at 
- Rewrite Necessary/GrowthGetters/asm/Write_Growths_To_Battle_Struct.s using the C decomp function as a base
- Add bracketed stats that stay within X points of the average, preventing becoming stat screwed and/or stat blessed, depending on your config. 
- This does not account for stat boosters (and fixed growths never has), as it would use up extra ram. If this is a problem to you, then set PREVENT_WHEN_ABOVE_AVERAGE_BY_AMOUNT to a high number.

masterofcontroversy commented:
What clib is this supposed to be using? Mokha's decomp compatible version and Stan's version are both implied to be used.

If Mokha's version is used, I think it'd be better to edit the makefile and definitions file to use the local installation of Mokha's clib to make compiling the code more straightforward in the future.

If Stan's version is used, why has Mokha's clib been locally added?

Veslyquix commented:
I just copied the header files from a folder of mine in pokemblem where I'd set things up earlier this year from the decomp. It should be similar to Mokha's setup I guess. 

Definitions file shouldn't need the 2019 file and the makefile just references devkitpro for lyn.exe - the include is just if you type `#include "gbafe.h"` at the start, I believe. So the 2019 stuff isn't used at all afaik.

sme23 commented on EngineHacks/Config.event:
```diff
@@ -359,11 +359,28 @@ MESSAGE Please see the README.md there or ask @Vesly for help.
 #define USE_STAT_COLORS True
 
+//When leveling up, the game rerolls (up to 5 times) until you gain at least this many points
+#define MIN_STAT_GAIN 2 // vanilla is 1 
```
should be the vanilla value by default

sme23 commented on EngineHacks/Config.event:
```diff
@@ -359,11 +359,28 @@ MESSAGE Please see the README.md there or ask @Vesly for help.
 //If true, Fixed Growths mode will exist.
-#define FIXED_GROWTHS_MODE False
+// Note: this takes promotion level into account, but does not take stat boosters into account 
+#define FIXED_GROWTHS_MODE True
```
should be off by default

sme23 reviewed (changes requested):
small config things

Vesly01 commented:
> If Mokha's version is used, I think it'd be better to edit the makefile and definitions file to use the local installation of Mokha's clib to make compiling the code more straightforward in the future.

I fixed the definitions.s file to not reference the 2019 version, but I'm not sure about how to edit the makefile to not have the `	$(CC) $(CFLAGS) -S $< -I C:/devkitPro/FE-CLib/include -o $@ -fverbose-asm -mlong-calls` stuff 

> small config things
Fixed this stuff
